packages/connect-core/src/utils/descriptor/describe.ts

```diff
@@ -0,0 +1,116 @@
+import { providers as ethersProviders } from 'ethers'
+
+import {
+  tryEvaluatingRadspec,
+  tryDescribingUpdateAppIntent,
+  postprocessRadspecDescription,
+} from '../radspec/index'
+import { StepDecoded, StepDescribed, PostProcessDescription } from '../../types'
+import App from '../../entities/App'
+import Transaction from '../../entities/Transaction'
+
+export async function describeStep(
```

Comment:
Here we describe a single Step in the path giving a `StepDecoded`. The `StepDecoded` could be a single step or have several children of `StepDecoded` representing further forwarding steps in the path.